src/lib/operators/abstract_operator.cpp

```diff
@@ -33,7 +35,14 @@ void AbstractOperator::execute() {
     _output = _on_execute(transaction_context);
     transaction_context->on_operator_finished();
   } else {
-    _output = _on_execute(transaction_context);
+    if (dynamic_cast<SQLQueryOperator*>(this)) {
+      // TODO(anyone): Once the SQL Query Operator is refactored (#304), this should be removed
+      _output = _on_execute(nullptr);
+    } else {
+      auto read_only_operator = dynamic_cast<AbstractReadOnlyOperator*>(this);
+      DebugAssert(read_only_operator, "Trying to execute read write operator without transaction context");
```

Comment:
This is already taken care of in the `AbstractReadWriteOperator::_on_execute()`

Reply:
Good point, removed.